pkg/sidecar/sidecar.go

```diff
@@ -93,25 +96,24 @@ func Run(version string) int {
 	// also, but they should be made to fail until the driver is validated.
 
 	// run grpc server until terminated
-	server, err := grpc.NewServer(
-		rt.KubeClient,
-		rt.CBTClient,
-		rt.DriverName,
-		grpc.ServerConfig{
-			Port:        *port,
-			TLSCertFile: *tlsCert,
-			TLSKeyFile:  *tlsKey,
-		})
+	server, err := grpc.NewServer(grpc.ServerConfig{Runtime: rt})
 	if err != nil {
 		klog.Fatalf("Failed to start GRPC server: %v", err)
 	}
-	go server.Start()
+
+	// Start the gRPC server. This call does not block but
+	// arranges for the sidecar health to be exposed.
+	// Metadata requests are not served until the driver is ready.
+	server.Start()
 
 	// check for a compatible CSI driver.
 	if err := rt.WaitTillCSIDriverIsValidated(); err != nil {
 		klog.Error(err)
 		return 1
 	}
 
+	// Block until the gRPC server terminates.
```

Comment:
Note: In the next PR for server health I plan is to notify the `server` that the CSI driver is ready just above this line.